flow/connectors/mysql/qvalue_convert.go

```diff
@@ -223,8 +223,81 @@ func processGeometryData(data []byte) qvalue.QValueGeometry {
 			return qvalue.QValueGeometry{Val: wkt}
 		}
 	}
-	strVal := string(data)
-	return qvalue.QValueGeometry{Val: strVal}
+	return qvalue.QValueGeometry{Val: string(data)}
+}
+
+// https://dev.mysql.com/doc/refman/8.4/en/time.html
+func processTime(str string) (time.Time, error) {
+	abs, isNeg := strings.CutPrefix(str, "-")
+	tpart, frac, _ := strings.Cut(abs, ".")
+
+	var nsec uint64
+	if len(frac) > 0 {
+		fint, err := strconv.ParseUint(frac, 10, 64)
+		if err != nil {
+			return time.Time{}, err
+		}
+		switch len(frac) {
```

Comment:
how about something like?

```
if len(frac) <= 9 {
    nsec = fint * int(math.Pow10(9 - len(frac)))
} else {
    nsec = fint
}
```

Reply:
I was going to complain about `math.Pow` perf overhead, but now I see `math.Pow10` is good